### PR TITLE
[AF-1775-6.5] SSH Ciphers and MAC configured via args on startup

### PR DESCRIPTION
**Pull request by desmax74**
@porcelli

docs: https://github.com/kiegroup/kie-docs/pull/1343

### CODE REVIEW

**Review by porcelli — APPROVED** [review]

**Comment by desmax74** [comment]
@ederign  add domhanak here

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProvider.java [hunk 1]
```diff
@@ -256,6 +258,8 @@ private void loadConfig( final ConfigProperties config ) {
         final ConfigProperty httpProxyPasswordProp = config.get( "http.proxyPassword", null );
         final ConfigProperty httpsProxyUserProp = config.get( "https.proxyUser", null );
         final ConfigProperty httpsProxyPasswordProp = config.get( "https.proxyPassword", null );
+        final ConfigProperty gitSshCipersConf = config.get("org.uberfire.nio.git.ssh.cyphers", null );
```
Comment (jhrcek) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProvider.java]: let's by consistent and use either cipher or cypher everywhere. Cipher is used in method names, why not use it in property name too? Just please don't forget to update the name of property in kie-docs if going for this change..

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 2]
```diff
@@ -166,31 +183,61 @@ public static SshServer buildSshServer() {
...
+            final byte[] key = new byte[c.getBlockSize()];
+            final byte[] iv = new byte[c.getIVSize()];
+            c.init(Cipher.Mode.Encrypt,
+                   key,
+                   iv);
+            ciphersHandled.add(cipher);
+        } catch (Exception e) {
+            LOG.warn("Cipher {} not handled in git ssh configuration, detail:{} ", cipher, e.getMessage());
+        }
+        LOG.info("Added Cipher {} to the git ssh configuration. ", cipher);
```
Comment (jhrcek) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: This `LOG.info` will be printed out even if Exception is thrown in the try block - the info log should be moved to line 218 at the end of the try block.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java [hunk 3]
```diff
@@ -166,31 +183,61 @@ public static SshServer buildSshServer() {
...
+            return managedMACs;
+        } else {
+            List<NamedFactory<Mac>> macs = new ArrayList<>();
+            List<String> macsInput = Arrays.asList(gitSshMacs.split(","));
+            for (String macCode : macsInput) {
+                NamedFactory<Mac> mac =  NamedFactory.Utils.get(managedMACs, macCode.trim().toLowerCase());
+                if (mac != null && managedMACs.contains(mac)) {
+                    macs.add(mac);
+                    LOG.info("Added MAC {} to the git ssh configuration. ", mac);
+                }else{
```
Comment (jhrcek) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Formatting - missing spaces before/after else

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java [hunk 4]
```diff
@@ -245,4 +262,215 @@ public void testCheckSetupParameters() throws Exception {
...
+                         macs);
+
+        sshService.start();
+        assertTrue(sshService.isRunning());
+
+        List<NamedFactory<Cipher>> ciphersReaded = sshService.getSshServer().getCipherFactories();
+        List<NamedFactory<Mac>> macsReaded = sshService.getSshServer().getMacFactories();
+
+        assertThat(ciphersReaded).hasSize(4);
+        assertThat(sshService.getManagedCiphers().containsAll(ciphersReaded));
```
Comment (jhrcek) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]: This assert does nothing - you must move the `)` to fix that. The same for assert on lines 294, 326, 329, and many others (ctrl+f `containsAll`)
```
assertThat(sshService.getManagedCiphers().containsAll(ciphersReaded));
->
assertThat(sshService.getManagedCiphers()).contains(ciphersReaded));
```

**Review by jhrcek — CHANGES_REQUESTED** [review]
Some logging and test fixes should be done, otherwise looks ok.
But I don't have any idea how to actually test this in the running workbench. Could you share some steps to check e.g. in git daemon of the running workbench that it's actually disabled? Or add steps to the AF-1775..

**Comment by jhrcek** [comment]
@desmax74 could you please address those comments on appformer/master too, so I don't have to duplicate them?

**Comment by desmax74** [comment]
@jhrcek sure I'll apply on the master too

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java (hunk 4 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]:
```
assertThat(sshService.getManagedCiphers().containsAll(ciphersReaded));
->
assertThat(sshService.getManagedCiphers()).containsAll(ciphersReaded));
```

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 3 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Fixed

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java (hunk 2 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java]: Fixed

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProvider.java (hunk 1 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProvider.java]: Fixed

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java (hunk 4 above)
Comment (desmax74) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java]: Fixed

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProvider.java [hunk 5]
```diff
@@ -256,6 +258,8 @@ private void loadConfig( final ConfigProperties config ) {
         final ConfigProperty httpProxyPasswordProp = config.get( "http.proxyPassword", null );
         final ConfigProperty httpsProxyUserProp = config.get( "https.proxyUser", null );
         final ConfigProperty httpsProxyPasswordProp = config.get( "https.proxyPassword", null );
+        final ConfigProperty gitSshCiphersConf = config.get("org.uberfire.nio.git.ssh.ciphers", null );
```
Comment (jhrcek) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProvider.java:261]: I actually meant the name of the system property should be changed to `org.uberfire.nio.git.ssh.ciphers`, but leaving the decision up to you.

**Review by jhrcek — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**